src/index.ts

```diff
@@ -0,0 +1,241 @@
+interface BinaryOperationToken {
+  type: 'BinaryOperationToken';
+  operation: '+' | '-' | '*' | '/';
+}
+
+interface ClosingParenthesis {
+  type: 'ClosingParenthesis';
+  value: ')';
+}
+
+interface NumberToken {
+  type: 'NumberToken';
+  value: number;
+}
+
+interface OpeningParenthesis {
+  type: 'OpeningParenthesis';
+  value: '(';
+}
+
+type NumberOrBinaryOperationToken = NumberToken | BinaryOperationToken;
+
+type Token =
+  | BinaryOperationToken
+  | NumberToken
+  | OpeningParenthesis
+  | ClosingParenthesis;
+
+const OPERATORS = new Set(['+', '-', '*', '/']);
+
+const operatorPrecedence = {
+  '*': 2,
+  '/': 2,
+  '+': 1,
+  '-': 1,
+};
+
+const VALID_CHARACTERS = new Set(['+', '/', '-', '*', '.', '(', ')']);
+
+for (let i = 0; i < 10; i++) {
+  VALID_CHARACTERS.add(String(i));
+}
+
+const isOperator = (element: string) => OPERATORS.has(element);
+const isNumber = (element: string) => /^-?\d+$/.test(element);
+
+const hasConsecutiveOperators = (string: string): boolean => {
```

Comment:
this would be a good place to document about why this is even necessary to check, because the function name itself doesn't explain the usefulness